addons/sale/views/sale_views.xml

```diff
@@ -233,18 +233,31 @@
                     </group>
                     <notebook>
                         <page string="Order Lines">
-                            <field name="order_line" mode="tree,kanban"
-                                attrs="{'readonly': [('state', 'in', ('done','cancel'))]}">
-                                <form string="Sales Order Lines">
+                            <field
+                                name="order_line"
+                                widget="section_and_note_one2many"
+                                mode="tree,kanban"
+                                attrs="{'readonly': [('state', 'in', ('done','cancel'))]}"
+                            >
+                                <form>
+                                    <field name="display_type" invisible="1"/>
+                                    <!--
+                                        We need the sequence field to be here for new lines to be added at the correct position.
+                                        TODO: at some point we want to fix this in the framework so that an invisible field is not required.
+                                    -->
+                                    <field name="sequence" invisible="1"/>
                                     <group>
-                                        <group>
+                                        <group attrs="{'invisible': [('display_type', '!=', False)]}">
+                                        <!-- TODO SEB: fix this not working -->
```

Comment:
What is not working ?

Reply:
I fixed it already, it was a remaining comment. Good catch, I'll remove the TODO.